online-shop/src/App.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import './App.css';
+import './productModeling/product.css';
+import './productModeling/productList.css';
+import PostData from './products.json';
+import ProductList from './components/ProductList';
+import ProductDetails from './components/ProductDetails';
+import './App.sass';
+import {  Switch, Route } from 'react-router';
+import {BrowserRouter as Router } from 'react-router-dom';
+import NavBar from './components/NavBar';
+import ShoppingCart from './components/ShoppingCart';
+import { library } from '@fortawesome/fontawesome-svg-core';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faStroopwafel } from '@fortawesome/free-solid-svg-icons';
+library.add(faStroopwafel)
+
+export interface IProduct{
```

Comment:
I generally recommend moving interfaces / classes which describe business data in a separate, `model` folder. In the `tsx` files, you should just have the component itself + props / state definition + any local helper functions.